src/libs/ck-libs/tcharm/tcharm_impl.h

```diff
@@ -248,7 +244,8 @@ class TCharm: public CBase_TCharm
 		if (tcharm_nothreads)
 			CkAbort("Cannot make blocking calls using +tcharm_nothreads!\n");
 		#endif
-		stopTiming();
+		// this should be triggered via array listeners?
+		// stopTiming();
```

Comment:
Has this been tested for `[threaded]` EPs? I think everything looks correct with regards to thread listeners, but it's a tricky enough area that I don't trust my own head.

Reply:
Yes, accurate object tracking surrounding `[threaded]` EPs is one of the primary motivations for this work. It has proven quite tricky to up-keep on my own, particularly after resuming from sleep!

Adding `CkEnforce(this == CkActiveObj());` around L151 and L158 of `tests/charm++/megatest/callback.C` will show it works correctly on entry, and after resume.